SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/trade/view/TradeViewModel.kt

```diff
@@ -0,0 +1,222 @@
+package app.cybrid.sdkandroid.components.trade.view
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import app.cybrid.cybrid_api_bank.client.apis.QuotesApi
+import app.cybrid.cybrid_api_bank.client.apis.TradesApi
+import app.cybrid.cybrid_api_bank.client.infrastructure.ApiClient
+import app.cybrid.cybrid_api_bank.client.models.*
+import app.cybrid.sdkandroid.AppModule
+import app.cybrid.sdkandroid.Cybrid
+import app.cybrid.sdkandroid.components.TradeView
+import app.cybrid.sdkandroid.components.listprices.view.ListPricesViewModel
+import app.cybrid.sdkandroid.core.AssetPipe
+import app.cybrid.sdkandroid.util.*
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+
+class TradeViewModel: ViewModel() {
+
+    private var quoteService = AppModule.getClient().createService(QuotesApi::class.java)
+    private var tradeService = AppModule.getClient().createService(TradesApi::class.java)
+
+    var uiState: MutableState<TradeView.ViewState> = mutableStateOf(TradeView.ViewState.LOADING)
+    var uiModalState: MutableState<TradeView.QuoteModalViewState> = mutableStateOf(TradeView.QuoteModalViewState.LOADING)
+
+    var currentFiatCurrency = "USD"
```

Comment:
Do we need this? Why is it hard coded?

Reply:
No is hardcoded! is the initial state, after this the ViewModel begin to play with states and change, but at the begining is always LOADING!